PyInstaller/lib/modulegraph/modulegraph.py

```diff
@@ -2339,6 +2339,24 @@ def _safe_import_hook(
         # Target module imported above.
         target_module = target_modules[0]
 
+        if (isinstance(target_module, MissingModule) and
+                source_module is not None and
+                target_attr_names is None and
+                level == ABSOLUTE_IMPORT_LEVEL and
+                type(source_module) is SourceModule and
+                target_module_partname ==
+                    '_' + source_module.identifier.rpartition('.')[2] and
+                sys.version_info[0] == 3):
+            # if this possible swig C module was previously imported from
+            # a python module other than its corresponding swig python
+            # module, then it may have been considering a MissingModule.
+            # Try to reimport it now
+            self.removeNode(target_module)
```

Comment:
I believe it is correct and I'll add a comment. My changes should just result in the same call to `_safe_import_hook` with an altered state in the module graph. If something seems fishy please make a big deal out of it since I'm only familiar with the part of the module graph that was an issue with _gdal and only became familiar with it over the last couple days.